### PR TITLE
AKS nodepool support

### PR DESCRIPTION
**Pull request by colin014**

### CODE REVIEW

File: components/azure/azure.go [hunk 1]
```diff
@@ -38,22 +39,18 @@ type ListResponse struct {
 }
 
 type CreateClusterAzure struct {
-	Node *CreateAzureNode `json:"node,omitempty"`
+	ResourceGroup     string                `json:"resourceGroup"`
+	KubernetesVersion string                `json:"kubernetesVersion"`
+	NodePools         *map[string]*NodePool `json:"nodePools,omitempty"`
```
Comment (stoader) [inline components/azure/azure.go]: Why do we need a pointer to map? map is a reference type already.

File: components/azure/azure.go [hunk 2]
```diff
@@ -38,22 +39,18 @@ type ListResponse struct {
...
+	VmSize     string `json:"vmSize"`
 }
 
 type UpdateClusterAzure struct {
-	*UpdateAzureNode `json:"node,omitempty"`
-}
-
-type UpdateAzureNode struct {
-	AgentCount int `json:"agentCount"`
+	NodePools *map[string]*NodePool `json:"nodePools,omitempty"`
```
Comment (stoader) [inline components/azure/azure.go]: Why do we need `*map` instead of `map`?

File: components/azure/azure.go [hunk 3]
```diff
@@ -69,26 +66,28 @@ func (azure *CreateClusterAzure) Validate() error {
 	}
 
 	// ---- [ Node check ] ---- //
-	if azure.Node == nil {
+	if azure.NodePools == nil {
 		msg := "Required field 'node' is empty."
```
Comment (stoader) [inline components/azure/azure.go]: The message refers to node instead of node pools.

File: components/azure/azure.go [hunk 4]
```diff
@@ -38,22 +39,18 @@ type ListResponse struct {
...
 }
 
-type CreateAzureNode struct {
-	ResourceGroup     string `json:"resourceGroup"`
-	AgentCount        int    `json:"agentCount"`
-	AgentName         string `json:"agentName"`
-	KubernetesVersion string `json:"kubernetesVersion"`
+type NodePool struct {
+	AgentCount int    `json:"agentCount"`
+	VmSize     string `json:"vmSize"`
```
Comment (stoader) [inline components/azure/azure.go]: This is a bit misleading. 
`nodePools: {
    "pool1" : {
        "agentCount":  1,
        "vmSize": "blah"
    }
}
`

We shouldn't mix agent and node pool terminology.

Comment (stoader) [inline components/azure/azure.go]: What is `vmSize`?

File: components/azure/azure.go [hunk 5]
```diff
@@ -69,26 +70,28 @@ func (azure *CreateClusterAzure) Validate() error {
 	}
 
 	// ---- [ Node check ] ---- //
-	if azure.Node == nil {
-		msg := "Required field 'node' is empty."
+	if azure.NodePools == nil {
```
Comment (baluchicken) [inline components/azure/azure.go]: Wrong Comment

**Pull request merged**